test/Interop/Cxx/constructor/copy-constructor-irgen.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: %target-swift-frontend -enable-cxx-interop -I %S/Inputs %s -emit-ir | %FileCheck %s
+
+import Basic
+
+// CHECK-LABEL: define swiftcc void @"$s4main4testSo30HasUserProvidedCopyConstructorVyF"
+// CHECK: [[OBJ:%.*]] = alloca %TSo30HasUserProvidedCopyConstructorV
+// Make sure we bitcast it from the Swift type to the C++ type.
+// CHECK: [[CXX_ARG:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* %0 to %struct.HasUserProvidedCopyConstructor*
+// CHECK: [[CXX_OBJ:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* [[OBJ]] to %struct.HasUserProvidedCopyConstructor*
+// Make sure we call the destructor.
+// CHECK: call void @_ZN30HasUserProvidedCopyConstructorC1ERKS_(%struct.HasUserProvidedCopyConstructor* [[CXX_ARG]], %struct.HasUserProvidedCopyConstructor* [[CXX_OBJ]])
+// CHECK: ret void
+public func test() -> HasUserProvidedCopyConstructor {
+  let obj = HasUserProvidedCopyConstructor(42)
+  return obj
+}
+
+// CHECK-LABEL: define linkonce_odr void @_ZN30HasUserProvidedCopyConstructorC1ERKS_
+// CHECK: call void @_ZN30HasUserProvidedCopyConstructorC2ERKS_
+
+// Make sure we store 42 into the pointer in the destructor
```

Comment:
s/destructor/copy constructor/

Also, this particular bit of the test isn't testing for 42 -- it's testing more generally that the argument gets stored into the object.

But again, I would suggest leaving this out entirely. This is testing Clang's codegen, which we really shouldn't be in the business of doing in a Swift test. a) It's unnecessary, b) it makes the test unnecessarily brittle: A (behavior-preserving) change in Clang's codegen could break this test.

Reply:
Done.